### ISSUE TITLE
Enabling haServices causes operator crash

### ISSUE THREAD

**Issue by sighup** [armory/spinnaker-operator#149]
Enabling haServices causes the operator container to panic and crash. We believe this may be due to the new deployment names that are that are created when enabling haServices for clouddriver. 

Operator version 1.0.2 in basic deployment.

```
        haServices:                                                                                                                                                                                                          
          clouddriver:
            disableClouddriverRoDeck: false
            enabled: true
          echo:
            enabled: false
```

```
E0728 17:15:39.111067       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:679
/usr/local/go/src/runtime/panic.go:199
/usr/local/go/src/runtime/signal_unix.go:394
/opt/spinnaker-operator/build/pkg/util/find.go:88
/opt/spinnaker-operator/build/pkg/deploy/spindeploy/transformer/accounts.go:92
/opt/spinnaker-operator/build/pkg/deploy/spindeploy/transformer/accounts.go:50
/opt/spinnaker-operator/build/pkg/deploy/spindeploy/spindeploy.go:93
/opt/spinnaker-operator/build/pkg/controller/spinnakerservice/spinnakerservice_controller.go:132
/opt/spinnaker-operator/build/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216
/opt/spinnaker-operator/build/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:192
/opt/spinnaker-operator/build/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:171
/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1357
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x268 pc=0x13f72d6]

goroutine 787 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1722360, 0x28b3d80)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/armory/spinnaker-operator/pkg/util.GetContainerInDeployment(0x0, 0x194cf3a, 0xb, 0x0)
	/opt/spinnaker-operator/build/pkg/util/find.go:88 +0x26
github.com/armory/spinnaker-operator/pkg/deploy/spindeploy/transformer.addSpringProfile(0x0, 0x194cf3a, 0xb, 0x19492cc, 0x8, 0x8010104, 0x0)
	/opt/spinnaker-operator/build/pkg/deploy/spindeploy/transformer/accounts.go:92 +0x5a
github.com/armory/spinnaker-operator/pkg/deploy/spindeploy/transformer.(*accountsTransformer).TransformManifests(0xc000b0bec0, 0x1cb0e40, 0xc0005302d0, 0xc0003a5260, 0xc0005c2278, 0x0, 0x0)
	/opt/spinnaker-operator/build/pkg/deploy/spindeploy/transformer/accounts.go:50 +0x110
github.com/armory/spinnaker-operator/pkg/deploy/spindeploy.(*Deployer).Deploy(0xc000117a40, 0x1cb0e40, 0xc0005302d0, 0x1cef720, 0xc000136280, 0xc0003a5260, 0x1e, 0x1961628, 0x1a)
	/opt/spinnaker-operator/build/pkg/deploy/spindeploy/spindeploy.go:93 +0x68d
github.com/armory/spinnaker-operator/pkg/controller/spinnakerservice.(*ReconcileSpinnakerService).Reconcile(0xc0002595e0, 0xc0005d36b0, 0xc, 0xc0005d3650, 0x9, 0xc00006ec00, 0x0, 0x0, 0x0)
	/opt/spinnaker-operator/build/pkg/controller/spinnakerservice/spinnakerservice_controller.go:132 +0x70f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00049ae60, 0x178e4c0, 0xc00082a460, 0x0)
	/opt/spinnaker-operator/build/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00049ae60, 0x0)
	/opt/spinnaker-operator/build/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:192 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00049ae60)
	/opt/spinnaker-operator/build/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:171 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00039abd0)
	/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00039abd0, 0x3b9aca00, 0x0, 0x1, 0xc00004c7e0)
	/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc00039abd0, 0x3b9aca00, 0xc00004c7e0)
	/opt/spinnaker-operator/build/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/opt/spinnaker-operator/build/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x32e
```

When ha is successfully enabled we typically see the following deployment names: 

spin-clouddriver-caching
spin-clouddriver-ro
spin-clouddriver-ro-deck
spin-clouddriver-rw

**Comment by ncknt** [comment]
Note that this should only happen when you turn account CRDs on with `spec.accounts.enabled: true`. The workaround is to leave it off. It only affects your ability to use `SpinnakerAccount`.

**Comment by aniketksh** [comment]
Hi @ncknt: According to the documentation, we have to set `spec.accounts.enabled: true` for SpinnakerService to use all SpinnakerAccount defined in the cluster. 
Doc link: https://github.com/armory/spinnaker-operator/blob/master/doc/options.md

**Comment by ncknt** [comment]
Hi @aniketksh : this is correct. `SpinnakerAccount` is still experimental though. The bug should be fixed and available in the upcoming 1.1.0. Thanks for the report!
